### PR TITLE
CIF-2566: Add missing datalayer events

### PR DESCRIPTION
**Pull request by laurentiumagureanu**
## Description

Added missing cmp:clicked component events

## Related Issue

CIF-2566

## Motivation and Context

Generate adobeDataLayer events

## How Has This Been Tested?

manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: react-components/src/utils/useDataLayerEvents.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+            if (item.bundle) {
+                eventInfo['bundle'] = true;
+            }
+
+            dataLayerUtils.pushEvent('cif:addToCart', eventInfo);
+        });
+    });
+
+    useEventListener(document, 'aem.cif.add-to-wishlist', async event => {
```
Comment (Buuhuu) [inline react-components/src/utils/useDataLayerEvents.js:37]: This change should be in a separate PR based on issue/CIF-2407.

**Review by Buuhuu — COMMENTED** [review]
What about product recommendation's open details? Do we need the click event there as well?

File: react-components/src/utils/useDataLayerEvents.js (hunk 1 above)
Comment (laurentiumagureanu) [reply react-components/src/utils/useDataLayerEvents.js:37]: not necessarily. it has no side-effects if the event is never triggered

Comment (Buuhuu) [reply react-components/src/utils/useDataLayerEvents.js:37]: In the base of this PR the even `aem.cif.add-to-wishlist` does not exist. for traceability it would be better to have this done as part of CIF-2407. However, if we merge both in the same release it is probably not worth the effort to move the change.

File: react-components/src/utils/useDataLayerEvents.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                'xdm:SKU': item.sku,
+                'xdm:quantity': item.quantity
+            };
+
+            dataLayerUtils.pushEvent('cif:addToWishList', eventInfo);
+        });
+    });
+};
+
+export default useDataLayerEvents;
```
Comment (herzog31) [inline react-components/src/utils/useDataLayerEvents.js:51]: What do you think about exporting this in the index as well, so it can be used directly without the need to use the `App` component?

**Review by herzog31 — APPROVED** [review]

File: react-components/src/utils/useDataLayerEvents.js (hunk 2 above)
Comment (laurentiumagureanu) [reply react-components/src/utils/useDataLayerEvents.js:51]: done 👍

**Review by Buuhuu — APPROVED** [review]
